Configuration/StandardSequences/python/earlyDeleteSettings_cff.py

```diff
@@ -122,11 +126,19 @@ def testHasInputTagModuleLabel(self):
                     refToPSet_ = cms.string("pset")
                 ),
             )
+            p.prod2 = cms.EDProducer("Producer2",
+                foo = cms.PSet(
+                    refToPSet_ = cms.string("nonexistent")
+                )
+            )
 
             result=[False,False,False,False,False,False,False,False,False,False,False,False,False,False]
-            _hasInputTagModuleLabel(p, p.prod, ["foo","foo2","foo3","bar","fred","wilma","a","foo4","bar2","bar3","fred2","wilma2","a2","joe"],result)
+            _hasInputTagModuleLabel(p, p.prod, "prod", ["foo","foo2","foo3","bar","fred","wilma","a","foo4","bar2","bar3","fred2","wilma2","a2","joe"], result)
             for i in range (0,13):
                 self.assert_(result[i])
             self.assert_(not result[13])
 
+            result = [False]
+            self.assertRaises(RuntimeError, _hasInputTagModuleLabel, p, p.prod2, "prod2", ["foo"], result)
```

Comment:
@makortel what would be the expected behaviour here? If I interpret correctly https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises I would expect an exception, running the unit test gives

```
15:04 cmsdev25 1679> python Configuration/StandardSequences/python/earlyDeleteSettings_cff.py
.
----------------------------------------------------------------------
Ran 1 test in 0.055s

OK
```

Reply:
If the function raises a `RuntimeError' exception, the `assertRaises()` succeeds. If the function does not raise an exception, or it raises an exception of some other type, the `assertRaises()` fails.